pkg/security/security.go

```diff
@@ -351,12 +352,41 @@ type Caller struct {
 	Identities []string
 }
 
+// AuthResult is the result of authentication action.
+type AuthResult struct {
+	Identities []string
+	Messages   []string
+}
+
+// AuthContext carries the contextual information and is passed across authenticators.
+type AuthContext struct {
+	// RequestContext is the context from request.
+	RequestContext context.Context
+	// Authenticators is the list of authenticators that were executed before in the auth chain.
+	Authenticators map[string]AuthResult
```

Comment:
I donot think this is a good struct, making the AuthContext aware of who may use it.

Reply:
Can you please exlain more on why it is not a good struct? AuthContext is a shared struct - to propagate the context of authenticators.

Reply:
The context should not contain `Authenticators ` 

context is passed to each authenticator, and it is a arg, not should be aware of which authenticator's decision. 

And I do not see the need to record who(authenticator) auth who (client) with what result, because a successful auth will make istiod break the loop.

Reply:
> context is passed to each authenticator, and it is a arg, not should be aware of which authenticator's decision.

That is for other authenticators to check which all authenticators have been passed unsuccessfully and what is their errors are. Successful authentication currently breaks the loop but before successful authenticator there can be failed authenticators that each authenticator can look at.

Reply:
>That is for other authenticators to check which all authenticators have been passed unsuccessfully and what is their errors are.

Why should be that?

Reply:
@costinm ^^